### PR TITLE
F1880 batch fargate support

### PR DESCRIPTION
**Pull request by ssickles**
added a new app type of batch:fargate, which can just be pushed to and view logs

### CODE REVIEW

File: aws/batch/deployer.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	fmt.Fprintf(stdout, "Deploying app %q\n", d.Details.App.Name)
+
+	jobDef, err := GetJobDefinition(ctx, d.Infra)
+	if err != nil {
+		return "", fmt.Errorf("error retrieving current job information: %w", err)
+	} else if jobDef == nil {
+		return "", fmt.Errorf("could not find job definition")
+	}
+
+	fmt.Fprintf(stdout, "DEBUG: version - %q\n", meta.Version)
```
Comment (BSick7) [inline aws/batch/deployer.go:59]: Did you intend to keep this or maybe drop the "DEBUG: " prefix?

Comment (ssickles) [reply aws/batch/deployer.go]: Just removed it.

File: aws/batch/update_job_definition.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		PlatformCapabilities: jobDefinition.PlatformCapabilities,
+		PropagateTags:        jobDefinition.PropagateTags,
+		RetryStrategy:        jobDefinition.RetryStrategy,
+		SchedulingPriority:   jobDefinition.SchedulingPriority,
+		Tags:                 jobDefinition.Tags,
+		Timeout:              jobDefinition.Timeout,
+	}
+	out, err := client.RegisterJobDefinition(ctx, input)
+	if err != nil {
+		return nil, err
```
Comment (BSick7) [inline aws/batch/update_job_definition.go:31]: Having additional context on this error result would be nice to distinguish it from Deregister.

File: aws/batch/update_job_definition.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	_, err = client.DeregisterJobDefinition(ctx, &batch.DeregisterJobDefinitionInput{
+		JobDefinition: &previousJobDefArn,
+	})
+	if err != nil {
+		return nil, err
```
Comment (BSick7) [inline aws/batch/update_job_definition.go:38]: Same as above.

Comment (ssickles) [reply aws/batch/update_job_definition.go]: Added more context to the error here.

File: aws/batch/update_job_definition.go (hunk 2 above)
Comment (ssickles) [reply aws/batch/update_job_definition.go]: Added more context to the error here.

**Pull request merged**